web_sdk/web_test_utils/lib/skia_client.dart

```diff
@@ -0,0 +1,399 @@
+  /// backend, the `add` argument uploads the current image test. A response is
+  /// returned from the invocation of this command that indicates a pass or fail
+  /// result for the tryjob.
+  ///
+  /// The [testName] and [goldenFile] parameters reference the current
+  /// comparison being evaluated.
+  Future<void> tryjobAdd(String testName, File goldenFile) async {
+    final List<String> imgtestCommand = <String>[
+      _goldctl,
+      'imgtest', 'add',
+      '--work-dir', _tempPath,
+      '--test-name', cleanTestName(testName),
+      '--png-file', goldenFile.path,
+    ];
+
+    final ProcessResult result = await process.run(imgtestCommand);
+
+    final String resultStdout = result.stdout.toString();
+    if (result.exitCode != 0 &&
+      !(resultStdout.contains('Untriaged') || resultStdout.contains('negative image'))) {
+      final StringBuffer buf = StringBuffer()
+        ..writeln('Unexpected Gold tryjobAdd failure.')
+        ..writeln('Tryjob execution for golden file test $testName failed for')
+        ..writeln('a reason unrelated to pixel comparison.')
+        ..writeln()
+        ..writeln('Debug information for Gold:')
+        ..writeln('stdout: ${result.stdout}')
+        ..writeln('stderr: ${result.stderr}')
+        ..writeln();
+      throw Exception(buf.toString());
+    }
+  }
+
+  /// Returns the latest positive digest for the given test known to Skia Gold
+  /// at head.
+  Future<String?> getExpectationForTest(String testName) async {
+    late String? expectation;
+    final String traceID = getTraceID(testName);
+    await HttpOverrides.runWithHttpOverrides<Future<void>>(() async {
+      final Uri requestForExpectations = Uri.parse(
+        '$_skiaGoldHost/json/v2/latestpositivedigest/$traceID'
+      );
+      late String rawResponse;
+      try {
+        final HttpClientRequest request = await httpClient.getUrl(requestForExpectations);
+        final HttpClientResponse response = await request.close();
+        rawResponse = await utf8.decodeStream(response);
+        final dynamic jsonResponse = json.decode(rawResponse);
+        if (jsonResponse is! Map<String, dynamic>)
+          throw const FormatException('Skia gold expectations do not match expected format.');
+        expectation = jsonResponse['digest'] as String?;
+      } on FormatException catch (error) {
+        print(
+          'Formatting error detected requesting expectations from Flutter Gold.\n'
+          'error: $error\n'
+          'url: $requestForExpectations\n'
+          'response: $rawResponse'
+        );
+        rethrow;
+      }
+    },
+      SkiaGoldHttpOverrides(),
+    );
+    return expectation;
+  }
+
+  /// Returns a list of bytes representing the golden image retrieved from the
+  /// Skia Gold dashboard.
+  ///
+  /// The provided image hash represents an expectation from Skia Gold.
+  Future<List<int>>getImageBytes(String imageHash) async {
```

Comment:
This method is not being used. The framework uses this for local testing, is it needed here?

Reply:
We are going to support local testing soon.